src/EditorFeatures/CSharp/Wrapping/CSharpArgumentWrappingCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,91 @@
+    {
+        protected override string ListName => FeaturesResources.argument_list;
+        protected override string ItemNamePlural => FeaturesResources.arguments;
+        protected override string ItemNameSingular => FeaturesResources.argument;
+
+        protected override SeparatedSyntaxList<ArgumentSyntax> GetListItems(BaseArgumentListSyntax listSyntax)
+            => listSyntax.Arguments;
+
+        protected override BaseArgumentListSyntax GetApplicableList(SyntaxNode node)
+        {
+            switch (node)
+            {
+                case InvocationExpressionSyntax invocationExpression: return invocationExpression.ArgumentList;
+                case ElementAccessExpressionSyntax elementAccessExpression: return elementAccessExpression.ArgumentList;
+                case ObjectCreationExpressionSyntax objectCreationExpression: return objectCreationExpression.ArgumentList;
+                case ConstructorInitializerSyntax constructorInitializer: return constructorInitializer.ArgumentList;
+            }
+
+            return null;
+        }
+
+        protected override bool PositionIsApplicable(
+            SyntaxNode root, int position,
+            SyntaxNode declaration, BaseArgumentListSyntax listSyntax)
+        {
+            var startToken = listSyntax.GetFirstToken();
+
+            // If we have something like  Foo(...)  or  this.Foo(...)  allow anywhere in the Foo(...)
+            // section.
+            if (declaration is InvocationExpressionSyntax ||
+                declaration is ElementAccessExpressionSyntax)
+            {
+                var expr = (declaration as InvocationExpressionSyntax)?.Expression ??
+                           (declaration as ElementAccessExpressionSyntax).Expression;
+                var name = expr as NameSyntax ??
+                           (expr as MemberAccessExpressionSyntax)?.Name ??
+                           (expr as MemberBindingExpressionSyntax)?.Name;
+
+                startToken = name == null ? listSyntax.GetFirstToken() : name.GetFirstToken();
+            }
+            else if (declaration is ObjectCreationExpressionSyntax objectCreation)
+            {
+                // allow anywhere in `new Foo(...)`
+                startToken = declaration.GetFirstToken();
+            }
+            else if (declaration is ConstructorInitializerSyntax constructorInitializer)
+            {
+                // allow anywhere in `this(...)` or `base(...)`
+                startToken = constructorInitializer.ThisOrBaseKeyword;
+            }
+
+            var endToken = listSyntax.GetLastToken();
+            var span = TextSpan.FromBounds(startToken.SpanStart, endToken.Span.End);
+            if (!span.IntersectsWith(position))
+            {
+                return false;
+            }
+
+            // allow anywhere in the arg list, as long we don't end up walking through something
+            // complex like a lambda/anonymous function.
```

Comment:
can you give an example of what you mean?

Reply:
I would imagine this would be available anywhere on `Foo` and then at the commas and parentheses, but not inside the arguments themselves (`$$` marks possible caret position)
```c#
Foo$$($$arg1$$,$$ arg2$$,$$ arg3$$)$$
```

Reply:
By example, i meant: a case where this would be not be a good experience.

I've been trying it out, and it feels ok to me :)